src/sort.ts

```diff
@@ -0,0 +1,94 @@
+/** @public */
+export type SortDirection =
+  | 1
+  | -1
+  | 'asc'
+  | 'desc'
+  | 'ascending'
+  | 'descending'
+  | { $meta: string };
+
+/** @public */
+export type Sort =
+  | string
+  | string[]
+  | { [key: string]: SortDirection }
+  | [string, SortDirection][]
+  | [string, SortDirection];
+
+/** Below stricter types were created for sort that correspond with type that the cmd takes  */
+
+/** @internal */
+type SortDirectionForCmd = 1 | -1 | { $meta: string };
+
+/** @internal */
+type SortForCmd = { [key: string]: SortDirectionForCmd };
+
+/** @internal */
+class SortDigest {
```

Comment:
Since all the methods are static so whats the motivation for making this a class?

Reply:
Since we haven't done it elsewhere I think its best to leave them as top level functions, the fact that they live in their own file I think already provides the boxing you're looking for.

Reply:
done!